laspy/_compression/backend.py

```diff
@@ -0,0 +1,83 @@
+import enum
+from abc import ABCMeta
+from typing import Tuple, BinaryIO, Any, Optional
+
+from ._laszipbackend import LaszipBackend
+from ._lazrsbackend import LazrsBackend
+from .lazbackend import ILazBackend
+from .selection import DecompressionSelection
+from .._pointreader import IPointReader
+from .._pointrecordwriter import IPointRecordWriter
+from .._pointwriter import IPointWriter
+
+_DEFAULT_BACKENDS: Tuple[ILazBackend, ...] = (
+    LazrsBackend(parallel=True),
+    LazrsBackend(parallel=False),
+    LaszipBackend(),
+)
+
+
+class ABCEnumMeta(enum.EnumMeta, ABCMeta):
+    pass
+
+
+# TODO: Replace the interactions with this class with interactions with ILazBackend classes
+class LazBackend(ILazBackend, enum.Enum, metaclass=ABCEnumMeta):
+    """Supported backends for reading and writing LAS/LAZ"""
+
+    # type_hint = Union[LazBackend, Iterable[LazBackend]]
+
+    LazrsParallel = 0
+    """lazrs in multi-thread mode"""
+    Lazrs = 1
+    """lazrs in single-thread mode"""
+    Laszip = 2
+    """laszip backend"""
+
+    def _get(self) -> ILazBackend:
+        return _DEFAULT_BACKENDS[self.value]
+
+    def is_available(self) -> bool:
+        """Returns true if the backend is available"""
+        for laz_backend in self.__class__:
+            laz_backend: LazBackend
+            if laz_backend == self:
+                return self._get().is_available()
+        return False
+
+    @property
+    def supports_append(self) -> bool:
+        return self._get().supports_append
+
+    def create_appender(
+        self, dest: BinaryIO, header: "LasHeader"
+    ) -> IPointRecordWriter:
```

Comment:
create_appender that retuns a IPointRecordWriter, feels a bit weird :thinking:

Reply:
If I change the name of this interface to `IPointAppender`, another problem will arise: `IPointAppender` will have a `write_points` method. Do you find it better? Do you have another suggestion?